### PR TITLE
Feat/send events

### PR DESCRIPTION
**Pull request by FredrikOseberg**
Adds capabilities to listen on events when doing the isEnabled and getVariant calls. These are emitted through the event listener, and can be picked up by the client and sent to any analytics provider.

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -170,30 +191,43 @@ export class UnleashClient extends TinyEmitter {
         }
     }
 
+    public createCustomEvent(toggleName: string, action: string) {
+        return this.eventsHandler.createCustomEvent(
+            this.context,
+            toggleName,
```
Comment (ivarconr) [inline src/index.ts:197]: toggleName needs to be optional.

**Comment by ivarconr** [comment]
Looks good. I think we can work with this. We should just create an alpha release directly from this branch.

File: src/index.ts [hunk 2]
```diff
@@ -170,30 +191,43 @@ export class UnleashClient extends TinyEmitter {
         }
     }
 
+    public createCustomEvent(action: string, toggleName?: string) {
+        return this.eventsHandler.createCustomEvent(
+            this.context,
+            action,
+            toggleName
+        );
```
Comment (ivarconr) [inline src/index.ts:199]: this needs to be emitted?

File: src/index.ts [hunk 3]
```diff
@@ -191,12 +192,13 @@ export class UnleashClient extends TinyEmitter {
         }
     }
 
-    public createCustomEvent(action: string, toggleName?: string) {
-        return this.eventsHandler.createCustomEvent(
+    public sendCustomEvent(toggleName: string) {
+        const event = this.eventsHandler.createCustomEvent(
             this.context,
-            action,
```
Comment (FredrikOseberg) [inline src/index.ts]: We should keep the action.

File: src/index.ts [hunk 4]
```diff
@@ -191,12 +192,13 @@ export class UnleashClient extends TinyEmitter {
...
             toggleName
         );
+        this.eventsHandler.addEvent(event);
```
Comment (FredrikOseberg) [inline src/index.ts]: We can probably also remove this for now.

**Review by chriswk — COMMENTED** [review]
Why the sudden change to double quotes in src/index.ts ?

**Comment by FredrikOseberg** [comment]
> Why the sudden change to double quotes in src/index.ts ?

There is a war between different editors going on right now.

**Pull request closed without merging**